packages/nimble-components/src/table/specs/table-column-width-hld.md

```diff
@@ -0,0 +1,170 @@
+
+### Out of Scope
+
+There are some column sizing behaviors that we will ultimately expect to support, but the APIs presented here are not meant to address:
+- Auto-resizing: We will not describe how we intend to support the use-case of having a column auto-size to its contents
+- Different interactive sizing modes: While the APIs described in this HLD do not inherently prescribe to a particular interactive sizing behavior, it's worth saying that in order to support multiple sizing modes, there will likely be additional APIs required that this HLD does not address.
+- Mechanisms related to accessibility-centric interactive column sizing (if there are such mechanisms). One possible example is allowing a user to size a column by way of the keyboard, instead of using a mouse. Ultimately, such a scenario is not in conflict with the API presented here, nor the mouse-based approach we know we will require, and can thus be handled separately, if ever.
+
+### API
+
+To accomodate the various sizing modes of a column, in addition to the other necessary behaviors, we can add the following properties to `TableColumn`:
+
+```
+class TableColumn {
+    // when set 'currentFractionalWidth' should be ignored
+    @observable
+    public currentPixelWidth: number | null = null;
+
+    @observable
+    public currentFractionalWidth = 1;
+
+    @observable
+    public currentMinWidth = 88; // the minimum size according to the design doc
+
+    @observable
+    public currentDisableResize = false;
+
+    ...
+}
+```
+Note that these properties are not attributes, and thus are not set by clients on the components via markup.
+
+These properties are expected to be updated by both the initial state of separate, column-specific APIs and interactive operations that the `Table` would manage.
+
+#### `currentFractionalWidth` vs `currentPixelWidth` behavior
+
+- The values supplied to the `currentFractionalWidth` property have the same meaning as the values supplied with the [`fr` unit](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout/Basic_Concepts_of_Grid_Layout#the_fr_unit) for the CSS `grid-template-columns` property.
+- The values supplied to the `currentPixelWidth` property are meant to be pixel-based values.
+- The table will use `currentPixelWidth` over `currentFractionalWidth` when both are set.
+
+#### `currentMinWidth` behavior
+
+Open Question: Is there a minimum width considered too small? If a user is allowed to provide a mimimum width smaller than the default, how should the various components within the header behave? If a user attempts to set a minimum width to an unsupported value, what is the appropriate response (i.e. should we simply update the `TableValidity` state?)?
+
+#### **Mixin Example**
+
+We can help facilitate proper implementation for concrete column types by providing mixins that define expected attribute APIs:
```

Comment:
Idle thought: we'd probably recommend that custom columns that want to be fractional width / resizable should always use this mixin. It would be poor form to have a fractional width column that controls its own width and doesn't expose a `fractionalWidth` attribute for the client to set. This is because setting fractional widths is a cooperative exercise across columns, so if a "greedy" column claimed it always needed 100fr then it would be hard for the client to know what `fractionalWidth` values to set for other columns.

No action needed here, just me wrapping my head around the API.